### PR TITLE
Refactoring accept/reject logic

### PR DESCRIPTION
**Pull request by Coder-Aht007**

### CODE REVIEW

File: donation/serializers.py [hunk 1]
```diff
@@ -61,3 +33,49 @@ def create(self, validated_data):
             search_slug=self.switch(validated_data['blood_group'])
         )
         return donation_request
+
+    def update(self, instance, validated_data):
+        instance.blood_group = validated_data.get(
```
Comment (mraarif) [inline donation/serializers.py:38]: why are we manually attaching related objects?

File: donation/serializers.py [hunk 2]
```diff
@@ -61,3 +33,49 @@ def create(self, validated_data):
...
+        instance.document = validated_data.get('document', instance.document)
+        if(validated_data.get('is_complete', False) is True):
+            instance.in_progress = False
+        instance.is_complete = validated_data.get(
+            'is_complete', instance.is_complete)
+        instance.save()
+        return instance
+
+
+class DonationGetSerializer(serializers.ModelSerializer):
```
Comment (mraarif) [inline donation/serializers.py:54]: why this name? this doesn't give me an idea, name should tell exactly why we need this

File: donation/serializers.py (hunk 1 above)
Comment (Coder-Aht007) [reply donation/serializers.py:49]: I didn't see any other approach except manually attaching logged in user to the request through serializers.........

File: donation/views.py [hunk 3]
```diff
@@ -17,52 +32,167 @@ class CustomPageNumberPagination(PageNumberPagination):
...
+        try:
+            response = sendGridClient.client.mail.send.post(
+                request_body=mail.get())
+            print(response)
+        except exceptions.BadRequestsError as e:
+            print("INSIDE")
+            print(e.body)
+
+    @staticmethod
+    @receiver(signals.post_save, sender=DonationRequest)
```
Comment (mraarif) [inline donation/views.py:109]: signals shouldn't be in views, there should be a separate file to store these handlers

File: donation/views.py [hunk 4]
```diff
@@ -17,52 +32,167 @@ class CustomPageNumberPagination(PageNumberPagination):
...
+            template_id, sender, recepient_email, data_dict)
+
+    @staticmethod
+    def send_email_to_requestor(data):
+        recepient_email = data.created_by.email
+        subject = "Donation Request Update"
+
+        body = {}
+        body['username'] = data.created_by.username
+        body['donation_role'] = "Donor"
```
Comment (mraarif) [inline donation/views.py:158]: do we need to copy over values like this? why can't we just send `data` as a context

File: donation/views.py [hunk 5]
```diff
@@ -17,52 +32,167 @@ class CustomPageNumberPagination(PageNumberPagination):
...
+        body['donation_role'] = "Requestor"
+        body['related_name'] = data.created_by.username
+        body['phone_number'] = data.created_by.phone_number
+        body['email'] = data.created_by.email
+        body['blood_group'] = data.blood_group
+        body['quantity'] = data.quantity
+        body['priority'] = data.priority
+        body['location'] = data.location
+
+        template_id = "d-4f03429f3f1a4db5a1aa80c02e563004"
```
Comment (mraarif) [inline donation/views.py:145]: instead of repeating it here why don't we create a service out of this

**Pull request merged**